google/auth/_helpers.py

```diff
@@ -88,3 +89,30 @@ def from_bytes(value):
     else:
         raise ValueError(
             '{0!r} could not be converted to unicode'.format(value))
+
+
+def update_query(url, params):
+    """Updates a URL's query parameters
+    Replaces any current values if they are already present in the URL.
+    Args:
+        url (str): The URL to update.
+        params (Mapping): A mapping of query parameter keys to values.
+    Returns:
+        str: The URL with updated query parameters.
+    """
+    # Split the URL into parts.
+    parts = urllib.parse.urlparse(url)
+    # Parse the query string.
+    query_params = urllib.parse.parse_qs(parts.query)
+    # Update the query parameters with the new parameters.
+    query_params.update(params)
+    # Remove any None values.
+    query_params = {
+        key: value for key, value
+        in six.iteritems(query_params)
+        if value is not None}
```

Comment:
Not really. I've updated the docstring with some examples - can you take a look and let me know how you feel?